plugin.video.BlanketFort/default.py

```diff
@@ -0,0 +1,515 @@
+profile = addon.getAddonInfo('profile').decode('utf-8')
+home = addon.getAddonInfo('path').decode('utf-8')
+favorites = xbmc.translatePath(os.path.join(profile, 'favorites' )).decode('utf-8')
+icon = xbmc.translatePath(os.path.join(home, 'icon.png'))
+fanart = xbmc.translatePath(os.path.join(home, 'fanart.jpg'))
+if os.path.exists(favorites)==True:
+    FAV = open(favorites).read()
+
+def makeRequest(url):
+        try:
+            req = urllib2.Request(url)
+            response = urllib2.urlopen(req)
+            data = response.read()
+            response.close()
+            return data
+        except urllib2.URLError, e:
+            if hasattr(e, 'code'):
+                print 'We failed with error code - %s.' % e.code
+                xbmc.executebuiltin("XBMC.Notification(BlanketFort,We failed with error code - "+str(e.code)+",10000,"+icon+")")
+            elif hasattr(e, 'reason'):
+                print 'We failed to reach a server.'
+                print 'Reason: ', e.reason
+                xbmc.executebuiltin("XBMC.Notification(BlanketFort,We failed to reach a server. - "+str(e.reason)+",10000,"+icon+")")
+
+
+def getSoup(url):
+        if url.startswith('http://'):
+            data = makeRequest(url)
+        else:
+            if xbmcvfs.exists(url):
+                if url.startswith("smb://"):
+                    copy = xbmcvfs.copy( url, xbmc.translatePath(os.path.join(profile, 'temp', 'sorce_temp.txt')))
+                    if copy:
+                        data = open( xbmc.translatePath(os.path.join(profile, 'temp', 'sorce_temp.txt')), "r").read()
+                        xbmcvfs.delete( xbmc.translatePath(os.path.join(profile, 'temp', 'sorce_temp.txt')) )
+                    else:
+                        print "--- failed to copy from smb: ----"
+                else:
+                    data = open(url, 'r').read()
+            else:
+                print "---- Soup Data not found! ----"
+                return
+        soup = BeautifulSOAP(data, convertEntities=BeautifulStoneSoup.XML_ENTITIES)
+        return soup
+
+
+def getData(url,fanart):
+        soup = getSoup(url)
+        if len(soup('channels')) > 0:
+            channels = soup('channel')
+            for channel in channels:
+                name = channel('name')[0].string
+                thumbnail = channel('thumbnail')[0].string
+                if thumbnail == None:
+                    thumbnail = ''
+
+                try:
+                    if not channel('fanart'):
+                        if addon.getSetting('use_thumb') == "true":
+                            fanArt = thumbnail
+                        else:
+                            fanArt = fanart
+                    else:
+                        fanArt = channel('fanart')[0].string
+                    if fanArt == None:
+                        raise
+                except:
+                    fanArt = fanart
+
+                try:

```

Comment:
Instead of all this try - except statements, just use .get(index, default_value).  something like desc = channel('info').get(0, '')